### ISSUE TITLE
Why are the running results different for the same data and parameter settings? 

### ISSUE THREAD

**Issue by qhdqhd** [nerfstudio-project/nerfstudio#1641]
![Uploading P~R{}OR@_[J95~G0F8)M9F2.png…]()
Psnr difference is 0.2dB.
Why are the running results different for the same data and parameter settings? Is it caused by random number seeds? How do I fix it?

**Comment by tancik** [comment]
We fix the random seeds here - https://github.com/nerfstudio-project/nerfstudio/blob/2c79d9fe52ae2eef2887aa621ae701ef91da95b0/scripts/train.py#L69-L73

Can you try adding `torch.cuda.manual_seed(seed)` to see if that helps.

**Comment by qhdqhd** [comment]
![}}@41H{%%VHVE{}D{U9XQT8](https://user-images.githubusercontent.com/42370719/227403771-a3c38afe-c914-466d-a4ae-54a21077ae97.png)
It does not work, the results are still different, confusing...

**Comment by tancik** [comment]
Take a look at other options here - https://pytorch.org/docs/1.13/notes/randomness.html?highlight=reproducability
For example maybe add `torch.use_deterministic_algorithms()`

It's possible that the non-determinism is coming from tinycudann, in which case I'm not sure how to get around it.
